src/gpu/intel/sdpa/micro.cpp

```diff
@@ -112,7 +112,11 @@ status_t micro_t::pd_t::init_conf_microkernels(impl::engine_t *engine) {
     /* Retrieve pre-tuned kernel configuration */
     config_t *config = nullptr;
     const dim_t thin_q_threshold = 16;
-    bool thin_q = (d->queries() <= thin_q_threshold);
+    auto queries = d->queries();
+    if (queries < thin_q_threshold)
+        queries *= (d->q_desc.dims[1] / d->kv_head_number);
```

Comment:
This condition looks different than the q == 1 condition in the kernel for enabling batching, if I understand correctly.

Reply:
Yeah, this condition helps to select the ideal tile shape for thin_q and first token cases. I was thinking that the condition was sufficient to handle both first and second token cases, but this condition could cause problems if the number of queries is less than `thin_q_threshold` which would select a less than ideal tile shape. 

I will adjust the condition to be more specific to second token cases.